src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp

```diff
@@ -101,7 +101,6 @@ void ObstacleAvoidance::injectAvoidanceSetpoints(Vector3f &pos_sp, Vector3f &vel
 	if (avoidance_data_timeout || !avoidance_point_valid) {
 		PX4_WARN("Obstacle Avoidance system failed, loitering");
 		_publishVehicleCmdDoLoiter();
-		return;
 	}
 
 	pos_sp = _sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].position;
```

Comment:
If you remove the `return;` you're injecting the avoidance waypoint which is not valid.
1. It's dangerous and bad practice to take over a point that is flagged invalid because it's misleading.
2. What does actually happen when you take the invalid point out of the topic? Is it zero? Does it stay on the last output? Would you apply constant velocity then?
It needs to be defined what happens and consistent with the naming as good as possible.